pkg/client/directory.go

```diff
@@ -142,11 +143,20 @@ func (d *directoryProvider) StatFile(_ context.Context, s string) (FileInfo, err
 		return FileInfo{}, err
 	}
 
+	// Get Mimetype
+	var mime string
+	mt, err := mimetype.DetectReader(f)
```

Comment:
This returns `application/octet-stream` when it fails to identify the mime type. I think we may want some logic to increase the read limit -- as per the `mimetype` [README](https://github.com/gabriel-vasile/mimetype?tab=readme-ov-file#faq) -- and try again when that happens.

Reply:
Yeah.. I don't know how sophisticated we want to get here, as for the current use case this is good enough.
The mentioned office documents are included by extension on knowledge side, so we won't care about the mimetype.
But I agree, that we need to have some more sophisticated approach here if we want to be 100% sure.

Reply:
That's fair enough!